File: Backend/src/main/java/com/dunelect/v1/service/MeterService.java
package com.dunelect.v1.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.dunelect.v1.models.Meters;

@Service
public interface MeterService {

	Meters getMeterById(int id);

	List<Meters> getAllMeters();

	Meters createMeter(Meters meter);
	
	Meters updateMeter(Meters meter,int id);
	
	void deleteMeter(int id);
}

File: Frontend/src/app/services/meter.service.ts
import { Injectable } from '@angular/core';
import { api } from './api.service';

@Injectable({ providedIn: 'root' })
export class MeterService {
  async getAllMeters() {
    const res = await api.get('/meters');
    return res.data;
  }
  async getMeterById(id: number) {
  const res = await api.get(`/meters/${id}`);
  return res.data;
}
  async createMeter(data: any) {
    const payload = {
      meterCode: data.meterCode,
      status: data.status,
      line: { lineId: data.lineId }, 
      createdAt: new Date().toISOString()
    };
    return (await api.post('/meters', payload)).data;
  }

  async updateMeter(id: number, data: any) {
    const payload = {
      meterId: id,
      meterCode: data.meterCode,
      status: data.status,
      line: { lineId: data.lineId },
      createdAt: data.createdAt 
    };
    return (await api.put(`/meters/${id}`, payload)).data;
  }

  async deleteMeter(id: number) {
    return (await api.delete(`/meters/${id}`)).data;
  }
}